### DISCUSSION TITLE
请问静态页面国际化预渲染建议方法?

### DISCUSSION THREAD

**Discussion by antaiSlasify** [umijs/umi#6648] [Q&A]
1.使用需求:
一个静态首页，在无渲染伺服器的情况下，使用预渲染机制(exportStatic)，生成英文、简中的静态档案，在布署dist档案夹到Netlify平台上，可透过URL 在https://. ../en, .../cn方式得到对应语系的画面，在开发模式下(F12)所见的也是对应的语系，非预设语系。
2. 目前尝试所碰到的问题:
(1) 当baseNavigator(config  folder>> config.ts)设为true，在英文与简中语系的浏览器上，SSR只会显示默认的简中语系 
 ssr: {},
 exportStatic: {}, //pre-render
 locale: {
    default zh-CN
    antd: true,
    // default true, when it is true, will use `navigator.language` overwrite default
    baseNavigator: false,
  },
(2) 在碰到上述(1)问题，改成使用router /en /cn的方法，想在match path去设定语系，但setLocale好像会用到浏览器语法，在SSR编译时会有错误 
![image](https://user-images.githubusercontent.com/84501228/119959090-3ea77300-bfd6-11eb-8651-4bd768cae5ac.png)
![image](https://user-images.githubusercontent.com/84501228/119959479-9e058300-bfd6-11eb-9a66-65e93f478ba5.png)
![image](https://user-images.githubusercontent.com/84501228/119959899-fd639300-bfd6-11eb-8f0d-d0517e1786a6.png)

想请问各位前辈，在这样的使用需求下，应该如何去做会比较好，谢谢

**Comment by knota-lab** [comment]
请问你解决了吗?

**Reply by antaiSlasify** [reply]
> 
没

**Reply by knota-lab** [reply]
我用原本的 react-intl `IntlProvider` 解决了，你可以试试
